### DISCUSSION TITLE
How to list all engines?

### DISCUSSION THREAD

**Discussion by HaoYang670** [apache/kyuubi#4993] [General]
The current `listEngine` admin API can't list all engines. Is there another way to achieve it?

I find there is a comment said Kyuubi doesn't support it, is it true?

https://github.com/apache/kyuubi/pull/4674#discussion_r1161379835

cc @zwangsheng  @pan3793

**Comment by zwangsheng** [comment]
Yes, for now you can't list all engines, and you can only list surviving engines.

Welcome to contribute REST-related interfaces, you can raise an issue describing requirements and features, wait for others to solve or directly provide PR.

**Reply by HaoYang670** [reply]
Field https://github.com/apache/kyuubi/issues/4994

**Reply by zwangsheng** [reply]
Thanks, would you like to push a PR to fix this?

**Reply by HaoYang670** [reply]
I mention in the issue that I may don't have time to do it. Sorry.

**Reply by pan3793** [reply]
@HaoYang670 Yes, the REST API and web UI are naive now. Anyway, thank you for paying attention to this area.

**Reply by zwangsheng** [reply]
Don't be sorry, if you don't have time Kyuubi community will arrange to complete this. Thanks for your report. @HaoYang670

**Reply by HaoYang670** [reply]
Thank you, @zwangsheng. We do need this feature.
